WalletWasabi.Gui/Controls/WalletExplorer/CoinJoinTabViewModel.cs

```diff
@@ -132,6 +131,10 @@ public CoinJoinTabViewModel(WalletViewModel walletViewModel)
 
 		private Global Global { get; }
 
+		private WalletService WalletService { get; }
+
+		private CompositeDisposable Disposables { get; set; }
```

Comment:
Something to consider: adding this and the whole dispose logic into WasabiDocumentTabViewModel OnOpen OnClose?

Reply:
Yes I have a better way of doing this, OnOpen is passed a disposable from the base class, and you dont need to calls Dispose on it, because when OnClose is called base class will dispose it after. Then we dont need to keep implementing this disposable pattern over and over again.

I'll do a seperate PR for that though.

Reply:
Yes I have a better way of doing this, OnOpen is passed a disposable from the base class, and you dont need to calls Dispose on it, because when OnClose is called base class will dispose it after. Then we dont need to keep implementing this disposable pattern over and over again.

I'll do a seperate PR for that though.